service/src/main/java/bio/terra/externalcreds/services/TokenProviderService.java

```diff
@@ -79,4 +83,51 @@ public void logLinkCreation(
                     .orElse(AuditLogEventType.LinkCreationFailed))
             .build());
   }
+
+  public Optional<String> getProviderAccessToken(String userId, Provider providerName) {
+    // get linked account
+    var linkedAccount =
+        linkedAccountService
+            .getLinkedAccount(userId, providerName.toString())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        String.format(
+                            "No linked account found for user ID: %s and provider: %s. "
+                                + "Please go to the Terra Profile page External Identities tab "
+                                + "to link your account for this provider.",
+                            userId, providerName)));
+
+    // get client registration from provider client cache
+    var clientRegistration =
+        providerClientCache
+            .getProviderClient(providerName.toString())
+            .orElseThrow(
+                () ->
+                    new ExternalCredsException(
+                        String.format(
+                            "Unable to find configs for the provider: %s", providerName)));
+
+    // make sure refresh token is populated in the linked account
+    if (linkedAccount.getRefreshToken().isEmpty()) {
+      throw new NotFoundException(
+          String.format("No refresh token found for provider %s", providerName));
+    }
+
+    // exchange refresh token for access token
+    var accessTokenResponse =
+        oAuth2Service.authorizeWithRefreshToken(
+            clientRegistration, new OAuth2RefreshToken(linkedAccount.getRefreshToken(), null));
+
+    // save the linked account with the new refresh token to replace the old one
+    var linkedAccountWithRefreshToken =
```

Comment:
`linkedAccountWithRefreshToken` is never used, so it doesn't need to be saved as a `var`